### DISCUSSION TITLE
Adding text based labels for atoms in sisl.viz?

### DISCUSSION THREAD

**Discussion by wladerer** [zerothi/sisl#817] [Q&A] [answered]
Is there a way to add labels to atoms using sisl.viz? I would like to be able to annotate structures with text, if possible. Thanks!

**Comment by zerothi** [comment]
@pfebrer have you seen this? What's the best way to do that?

**Reply by pfebrer** [reply, marked as answer]
Hey! I hadn't seen it. There is no way currently. I had thought of adding a `annotation` key to `atoms_style`, but I never did.

I think it would be a good addition (and also for `bonds_style`), but unfortunately I'm now writing my thesis on a really tight schedule and I have no time to do anything.

@wladerer are you using the plotly backend? Then probably the best you can do is to update the atoms trace with an extra `text` attribute. If you don't know how to do it let me know and I'll prepare a quick code snippet :)

**Reply by wladerer** [reply]
I am. Big brain move. I will try that out! 

Best of luck with your thesis stuff.

**Reply by pfebrer** [reply]
> Best of luck with your thesis stuff.

Thanks!

This is the approach I follow in general when I need something that is not supported by sisl.viz. That's why I love plotly, it is easy to update everything a posteriori.
